### PR TITLE
added offline mode (service workers)

### PR DESCRIPTION
**Pull request by KalaRaja**
'check for updates' polling interval is 7 minutes.

someone with a linux machine please check if below script works, doesnt work in windows because windows has DOS shell, not unix bash.

`"serve:build": "npm run build:libs && npm run build && ./node_modules/http-server/bin/http-server -p 4200 -c-1 dist/make-a-vis-demo",`

### CODE REVIEW

File: src/app/services/app-updater.service.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import { SwUpdate } from '@angular/service-worker';
+import { concat, interval } from 'rxjs';
+import { first } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AppUpdaterService {
+  isSnackbarloaded = false;
+  checkUpdateInterval = 1000 * 60 * 7;
```
Comment (bherr2) [inline src/app/services/app-updater.service.ts]: 7 minutes is far too often. Let's do every 2 hours (which is still probably too much).

File: package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "scripts": {
     "ng": "node --max_old_space_size=3000 ./node_modules/.bin/ng",
     "start": "npm run ng serve",
+    "serve:build": "npm run build:libs && npm run build && ./node_modules/http-server/bin/http-server -p 4200 -c-1 dist/make-a-vis-demo",
```
Comment (bherr2) [inline package.json]: This doesn't need to do the building along with the serving. Just serve. Also you should be able to refer to http-server like this: `"serve:build": "http-server -p 4200 -c-1 dist/make-a-vis-demo"`.

File: src/app/services/app-updater.service.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  constructor(
```
Comment (bherr2) [inline src/app/services/app-updater.service.ts]: Follow up question. Does this do an update check on initial load as well? If not, it should.

File: package.json (hunk 2 above)
Comment (KalaRaja) [reply package.json]: but to refer http-server, we would have to npm install http-server globally

File: src/app/services/app-updater.service.ts (hunk 3 above)
Comment (KalaRaja) [reply src/app/services/app-updater.service.ts]: yes it does

**Pull request merged**